### PR TITLE
updated application to react 18

### PR DESCRIPTION
**Pull request by bhavya2611**
This PR moves our application to use the latest version of React. 

This also makes sure that we don't use any vulnerable libraries.

### CODE REVIEW

File: next.config.js [hunk 1]
```diff
@@ -1,19 +1,19 @@
...
-    return config
-  },
-})
-
-module.exports = nextConfig
+/** @type {import('next').NextConfig} */
+const nextConfig = {
+  reactStrictMode: true,
+  images: {
+    loader: 'akamai',
```
Comment (neekolas) [inline next.config.js:5]: What's this about an Akamai loader? Why do we need any sort of third party service involved with our image loading?

File: components/Layout.tsx [hunk 2]
```diff
@@ -59,16 +42,21 @@ const Layout: React.FC = ({ children }) => {
...
-            <NavigationHeaderLayout>
-              {walletAddress && client && <NewMessageButton />}
-            </NavigationHeaderLayout>
-            <NavigationPanel onConnect={handleConnect} />
-            <UserMenu
-              onConnect={handleConnect}
-              onDisconnect={handleDisconnect}
-            />
-          </NavigationColumnLayout>
+          <aside className="flex w-full md:w-84 flex-col flex-grow fixed inset-y-0">
```
Comment (neekolas) [inline components/Layout.tsx:45]: I'm not sure what this has to do with upgrading to React 18

File: components/AddressInput.test.tsx [hunk 3]
```diff
@@ -2,26 +2,33 @@ import React from 'react'
...
+    waitFor(() => expect(input).toHaveAttribute('value', ''))
   })
 
   it('renders initial value', () => {
+    let input: HTMLInputElement | null
     act(() => {
       const { container } = render(
         <AddressInput recipientWalletAddress={'0xfoo.eth'} />
       )
-      expect(container.querySelector('input')).toHaveAttribute(
```
Comment (neekolas) [inline components/AddressInput.test.tsx:20]: Is this change required by the change to React 18, or did this just sneak in?

File: components/Conversation/MessageComposer.tsx [hunk 4]
```diff
@@ -58,12 +59,11 @@ const MessageComposer = ({ onSend }: MessageComposerProps): JSX.Element => {
...
         />
         <button type="submit" className={messageComposerStyles.arrow}>
-          <img
-            src={message ? upArrowGreen : upArrowGrey}
-            alt="send"
-            height={32}
-            width={32}
-          />
+          {!message ? (
+            <Image src={upArrowGrey} alt="send" height={32} width={32} />
```
Comment (neekolas) [inline components/Conversation/MessageComposer.tsx:63]: What advantages does the switch to Next Images give us? Are there any drawbacks?

File: next.config.js (hunk 1 above)
Comment (bhavya2611) [reply next.config.js:5]: This is I think built-in next and it is just there to compress images

https://nextjs.org/docs/api-reference/next/legacy/image#built-in-loaders

Added this as we used a loader before to optimize images as next does not optimize svg I believe

File: components/Layout.tsx (hunk 2 above)
Comment (bhavya2611) [reply components/Layout.tsx:45]: Umm I removed this as every component requires children prop explicitly in React 18, I saw no reason to have components to pass children around and cause reloads

File: components/AddressInput.test.tsx (hunk 3 above)
Comment (bhavya2611) [reply components/AddressInput.test.tsx:20]: Not very confident, but without waitFor it was failing and giving null reference for the loaded dom, I tweaked this therefore, maybe a react 18 thing, because on other PR's all of this worked

File: components/Conversation/MessageComposer.tsx (hunk 4 above)
Comment (bhavya2611) [reply components/Conversation/MessageComposer.tsx:63]: Image component uses optimization for images that's the only advantage, we can configure other things like cache time etc for these images also, but mainly to reduce the size of the image before serving is the reason to use this
